actors/builtin/miner/miner_state.go

```diff
@@ -259,6 +269,151 @@ func (st *State) ComputeProvingSet(store adt.Store) ([]abi.SectorInfo, error) {
 	return sectorInfos, nil
 }
 
+func computeVestAmount(total abi.TokenAmount, cliffStartEpoch abi.ChainEpoch, fullyVestedEpoch abi.ChainEpoch, quantizedEpoch abi.ChainEpoch) abi.TokenAmount {
+	// PARAM_FINISH linear vesting placeholder
+	numVestWindows := (fullyVestedEpoch - cliffStartEpoch) / VestIncrement
+	_ = (quantizedEpoch - cliffStartEpoch) / VestIncrement // number of VestIncrement elapsed for other vest functions
+	quantizedVestAmount := big.Div(total, big.NewInt(int64(numVestWindows)))
+	return quantizedVestAmount
+}
+
+func getQuantizedEpoch(e abi.ChainEpoch) abi.ChainEpoch {
+	// PARAM_FINISH
+	// default is the nearest next epoch that is multiple of QuantizedGranularity after a VestIncrement offset
+	// computeVestAmount can assume at least one VestIncrement has elapsed
+	offsetEpoch := e + VestIncrement
+	quantizedEpoch := offsetEpoch - (offsetEpoch % QuantizedGranularity) + QuantizedGranularity
+
+	return quantizedEpoch
+}
+
+func (st *State) addLockedFund(store adt.Store, currEpoch abi.ChainEpoch, pledgeAmount abi.TokenAmount) error {
+	Assert(pledgeAmount.GreaterThanEqual(big.Zero()))
+
+	vestingFunds := adt.AsArray(store, st.VestingFunds)
+
+	cliffStartEpoch := currEpoch + PledgeCliff
+	fullyVestedEpoch := currEpoch + PledgeCliff + PledgeVestingPeriod
+
+	// default vesting happens once a week at the quantized epoch.
+	for i := cliffStartEpoch; i < fullyVestedEpoch; i += VestIncrement {
+		lockedFundEntry := big.Zero()
+		quantizedVestEpoch := getQuantizedEpoch(i)
+		_, err := vestingFunds.Get(epochKey(quantizedVestEpoch), &lockedFundEntry)
+
+		if err != nil {
+			return err
+		}
+
+		vestAmount := computeVestAmount(pledgeAmount, cliffStartEpoch, fullyVestedEpoch, quantizedVestEpoch)
+		newLockedFundEntry := big.Add(lockedFundEntry, vestAmount)
+		vestingFunds.Set(epochKey(quantizedVestEpoch), &newLockedFundEntry)
+	}
```

Comment:
As from comments on computeVestAmount, this loop cannot be structured like this and still be correct in the face of integer division. It needs to be structured like:
```
while totalVested < total {
	x := next payment from (total, totalVested, remainingEpochs, vestIncrement)
    vest x
	totalVested += x
}
```